wallet/udb/vsp.go

```diff
@@ -73,9 +75,20 @@ func SetVSPTicket(dbtx walletdb.ReadWriteTx, ticketHash *chainhash.Hash, record
 			return err
 		}
 		record.VSPHostID = vspHostID
-	} else {
-		record.VSPHostID = pubkey.ID
+	} else if err != nil {
+		return err
+	}
+
+	// If the pubkey from the record in the request differs from the pubkey
+	// in the database that is saved for the host, update the pubkey in the
+	// db but keep the vsphost id intact.
+	if pubkey.PubKey != nil && bytes.Compare(pubkey.PubKey, record.PubKey) != 0 {
```

Comment:
`GetVSPPubKey` will return a `nil` `pubkey` if it returns `errors.NotExist` so this will panic in that case. You need to test whether `pubkey` is nil (not `pubkey.PubKey`). `bytes.Compare` can handle a nil `pubkey.PubKey` (though that won't happen in practice).

Reply:
ah good catch